app/grandchallenge/components/models.py

```diff
@@ -2221,15 +2221,35 @@ def remove_civ(self, *, civ):
         if not self.is_editable:
             raise RuntimeError(f"{self} is not editable.")
 
+    def validate_values_and_execute_linked_task(
+        self, *, values, user, linked_task=None
+    ):
+        if not self.is_editable:
+            raise RuntimeError(
+                f"{self} is not editable. No CIVs can be added or removed from it."
+            )
+        else:
+            for civ_data in values:
+                try:
+                    self.create_civ(
+                        civ_data=civ_data,
+                        user=user,
+                        linked_task=linked_task,
+                    )
+                except RuntimeError:
```

Comment:
Why do we catch this here? The same sort of error is raised above so callers are expected to handle this I think.

Additionally, catching `RuntimeError` might not be specific enough. It is being caught without being inspected, so could hide other `RuntimeError`s. Maybe a specific `CIVNotEditableException` should be created and used?

Reply:
A more specific exception is a good idea indeed. 

And as for catching it here: the editable state can change during the for-loop, so while the object is still editable when this function is called and also still for the first civ that is being created, it might no longer be when calling `create_civ` for the second time (because the first civ failed validation), hence why we catch it here.

Reply:
Indeed, but if the caller of `validate_values_and_execute_linked_task` is expected to handle `RuntimeError/CIVNotEditableException` which is raised on L2228 then we can let `self.create_civ` raise it too, so shouldn't need to explicitly handle it here, right? It also seems like `self.is_editable` is checked in `self.create_civ`, so the extra check here is unnecessary? This function would then reduce down to:

```python
    def validate_values_and_execute_linked_task(
        self, *, values, user, linked_task=None
    ):
            for civ_data in values:
                self.create_civ(
                    civ_data=civ_data,
                    user=user,
                    linked_task=linked_task,
                )
```

Reply:
Much clearer!